docs/src/main/sphinx/security/file-system-access-control.rst

```diff
@@ -1,34 +1,58 @@
-==========================
-File system access control
-==========================
+=========================
+File-based access control
+=========================
 
-This access control plugin allows you to specify authorization rules in a JSON file.
+To secure access to data in your cluster, you can implement file-based access
+control where access to data and operations is defined by rules declared in
+manually-configured JSON files.
+
+There are two types of file-based access control:
+
+* **System-level access control** uses the access control plugin with a single
+  JSON file that specifies authorization rules for the whole cluster.
+* **Catalog-level access control** uses individual JSON files for each catalog
+  for granular control over the data in that catalog, including column-level
+  authorization.
+
+File-based access control always prioritizes deny outcomes. If a rule in either
+a system-level or catalog-level access control file denies a particular
+operation, the operation is denied even if another rule is intended to allow
+that operation.
```

Comment:
I don't think this is accurate. AFAIK it's first matching rule wins.

i.e. if I have rule_allow and rule_deny (in that order) then even in the case where both rules are possible matches only rule_allow would be applied.

@kokosing can double check me on this (I looked in `io/trino/plugin/base/security/FileBasedSystemAccessControl.java`)

Reply:
That is correct. We process them from to top to bottom. If none match, then access is denied. If any match then it is allowed.

I don't recall any deny rules in file based access controls. @hashhar Can you please help to me find them? If they are there, then expected behavior is to evaluate deny rules first.

Reply:
I couldn't find any implementations either. There is SPI for it though.

Reply:
From my conversation with @electrum last Friday it sounded like a rule_deny anywhere would override a rule_allow anywhere. If that's wrong I'll update, but want to make sure I'm not misunderstanding.

Reply:
Updated the introduction, please take a look.